src/expandableCalendar/agendaListNew.tsx

```diff
@@ -0,0 +1,259 @@
+ */
+const AgendaList = (props: AgendaListProps) => {
+  const {
+    theme,
+    sections,
+    scrollToNextEvent,
+    avoidDateUpdates,
+    onScroll,
+    renderSectionHeader,
+    sectionStyle,
+    dayFormatter,
+    dayFormat = 'dddd, MMM d',
+    useMoment,
+    markToday = true,
+  } = props;
+
+  const {date, updateSource, setDate} = useContext(Context);
+
+  const style = useRef(styleConstructor(theme));
+  const list = useRef<any>();
+  const _topSection = useRef(sections[0]?.title);
+  const didScroll = useRef(false);
+  const sectionScroll = useRef(false);
+  const sectionHeight = useRef(0);
+  const [data, setData] = useState([] as any[]);
+
+  useEffect(() => {
+    const items = sections.reduce((acc: any, cur: any) => {
+      return [...acc, {title: cur.title, isTitle: true}, ...cur.data];
+    }, []);
+    setData(items);
+
+    if (date !== _topSection.current) {
+      setTimeout(() => {
+        scrollToSection(date);
+      }, 500);
+    }
+  }, [sections]);
+
+  useDidUpdate(() => {
+    // NOTE: on first init data should set first section to the current date!!!
+    if (updateSource !== UpdateSources.LIST_DRAG && updateSource !== UpdateSources.CALENDAR_INIT) {
+      scrollToSection(date);
+    }
+  }, [date]);
+
+  const getSectionIndex = (date: string) => {
+    let dataIndex = 0;
+
+    for (let i = 0; i < sections.length; i++) {
+      if (sections[i].title === date) {
+        return dataIndex;
+      }
+      dataIndex += sections[i].data.length + 1;
+    }
+  };
+
+  const getNextSectionIndex = (date: string) => {
+    const cur = new XDate(date);
+    let dataIndex = 0;
+
+    for (let i = 0; i < sections.length; i++) {
+      const titleDate = parseDate(sections[i].title);
+      if (isGTE(titleDate,cur)) {
+        return dataIndex;
+      }
+      dataIndex += sections[i].data.length + 1;
+    }
+  };
+
+  const getSectionTitle = useCallback((title: string) => {
+    if (!title) return;
+
+    let sectionTitle = title;
+
+    if (dayFormatter) {
+      sectionTitle = dayFormatter(title);
+    } else if (dayFormat) {
+      if (useMoment) {
+        const moment = getMoment();
+        sectionTitle = moment(title).format(dayFormat);
+      } else {
+        sectionTitle = new XDate(title).toString(dayFormat);
+      }
+    }
+
+    if (markToday) {
+      const string = getDefaultLocale().today || todayString;
+      const today = isToday(title);
+      sectionTitle = today ? `${string}, ${sectionTitle}` : sectionTitle;
+    }
+
+    return sectionTitle;
+  }, []);
+
+  const scrollToSection = useCallback(debounce((d) => {
+    const sectionIndex = scrollToNextEvent ? getNextSectionIndex(d) : getSectionIndex(d);
+    if (isUndefined(sectionIndex)) {
+      return;
+    }
+
+    if (list?.current && sectionIndex !== undefined) {
+      sectionScroll.current = true; // to avoid setDate() in onViewableItemsChanged
```

Comment:
You're not using onViewableItemsChanged here so why onViewableItemsChanged still set?